content/docs/setup/kubernetes/additional-setup/mesh-expansion/gateways/index.md

```diff
@@ -0,0 +1,459 @@
+$ curl -v httpbin.bar.global:8000
+< HTTP/1.1 200 OK
+< server: envoy
+< content-type: text/html; charset=utf-8
+< content-length: 9593
+
+... html content ...
+    {{< /text >}}
+
+The `server: envoy` header indicates that the sidecar intercepted the traffic.
+
+## Running services on a mesh expansion machine
+
+1. Setup an HTTP server on the VM instance to serve HTTP traffic on port 8080:
+
+    {{< text bash >}}
+    $ gcloud compute ssh ${GCE_NAME}
+    $ python -m SimpleHTTPServer 8888
+    {{< /text >}}
+
+1. Determine the VM instance's IP address.
+
+1. Configure a service entry to enable service discovery for the VM. You can add VM services to the mesh using a
+    [service entry](/docs/reference/config/networking/v1alpha3/service-entry/). Service entries let you manually add
+    additional services to Pilot's abstract model of the mesh. Once VM services are part of the mesh's abstract model,
+    other services can find and direct traffic to them. Each service entry configuration contains the IP addresses, ports,
+    and appropriate labels of all VMs exposing a particular service, for example:
+
+    {{< text bash yaml >}}
+    $ kubectl -n ${SERVICE_NAMESPACE} apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+      name: vmhttp
+    spec:
+      hosts:
+      - vmhttp.${SERVICE_NAMESPACE}.svc.cluster.local
+      ports:
+      - number: 8888
+        name: http
+        protocol: HTTP
+      resolution: STATIC
+      endpoints:
+      - address: ${GCE_IP}
+        ports:
+          http: 8888
+        labels:
+          app: vmhttp
+          version: "v1"
+    EOF
+    {{< /text >}}
+
+1. The workloads in a Kubernetes cluster need a DNS mapping to resolve the domain names of VM services. To
+    integrate the mapping with your own DNS system, use `istioctl register` and creates a Kubernetes `selector-less`
+    service, for example:
+
+    {{< text bash >}}
+    $ istioctl  register -n ${SERVICE_NAMESPACE} vmhttp ${VM IP} 8888
+    {{< /text >}}
+
+    {{< tip >}}
+    Make sure you have already added `istioctl` client to your `PATH` environment variable, as described in the Download page.
+    {{< /tip >}}
+
+1. Deploy a pod running the `sleep` service in the Kubernetes cluster, and wait until it is ready:
+
+    {{< text bash >}}
+    $ kubectl apply -f @samples/sleep/sleep.yaml@
+    $ kubectl get pod
+    NAME                             READY     STATUS    RESTARTS   AGE
+    productpage-v1-8fcdcb496-xgkwg   2/2       Running   0          1d
+    sleep-88ddbcfdd-rm42k            2/2       Running   0          1s
+    ...
+    {{< /text >}}
+
+1. Send a request from the `sleep` service on the pod to the VM's HTTP service:
+
+    {{< text bash >}}
+    $ kubectl exec -it sleep-88ddbcfdd-rm42k -c sleep -- curl vmhttp.${SERVICE_NAMESPACE}.svc.cluster.local:8888
+    {{< /text >}}
+
+    You should see something similar to the output below.
```

Comment:
You should/If configured properly, you will

Reply:
fixed